framework/wazuh/core/cluster/utils.py

```diff
@@ -18,15 +18,80 @@
 
 from wazuh.core import common, pyDaemonModule
 from wazuh.core.configuration import get_ossec_conf
-from wazuh.core.exception import WazuhException, WazuhError, WazuhInternalError
+from wazuh.core.exception import WazuhError, WazuhException, WazuhInternalError
 from wazuh.core.results import WazuhResult
 from wazuh.core.utils import temporary_cache
 from wazuh.core.wazuh_socket import create_wazuh_socket_message
 from wazuh.core.wlogging import WazuhLogger
 
+NO = 'no'
+YES = 'yes'
+DISABLED = 'disabled'
+HAPROXY_HELPER = 'haproxy_helper'
+HAPROXY_DISABLED = 'haproxy_disabled'
+HAPROXY_ADDRESS = 'haproxy_address'
+HAPROXY_PORT = 'haproxy_port'
+HAPROXY_PROTOCOL = 'haproxy_protocol'
+HAPROXY_USER = 'haproxy_user'
+HAPROXY_PASSWORD = 'haproxy_password'
+HAPROXY_BACKEND = 'haproxy_backend'
+HAPROXY_RESOLVER = 'haproxy_resolver'
+FREQUENCY = 'frequency'
+EXCLUDED_NODES = 'excluded_nodes'
+AGENT_CHUNK_SIZE = 'agent_chunk_size'
+AGENT_RECONNECTION_TIME = 'agent_reconnection_time'
+AGENT_RECONNECTION_STABILITY_TIME = 'agent_reconnection_stability_time'
+IMBALANCE_TOLERANCE = 'imbalance_tolerance'
+REMOVE_DISCONNECTED_NODE_AFTER = 'remove_disconnected_node_after'
+
 logger = logging.getLogger('wazuh')
 execq_lockfile = os.path.join(common.WAZUH_PATH, "var", "run", ".api_execq_lock")
 
+def validate_haproxy_helper_config(helper_config: dict) -> dict:
+    """Validate HAProxy helper configuration section.
+
+    Parameters
+    ----------
+    helper_config : dict
+        Configuration to validate.
+
+    Returns
+    -------
+    dict
+        Validated configuration for HAProxy Helper.
+
+    Raises
+    ------
+    WazuhError (3004)
+        If some value have an invalid type.
+    """
+    if helper_config[HAPROXY_DISABLED] == NO:
+        helper_config[HAPROXY_DISABLED] = False
+    elif helper_config[HAPROXY_DISABLED] == YES:
+        helper_config[HAPROXY_DISABLED] = True
+
+    for field in [
+        FREQUENCY,
+        AGENT_RECONNECTION_STABILITY_TIME,
+        AGENT_RECONNECTION_TIME,
+        AGENT_CHUNK_SIZE,
+        REMOVE_DISCONNECTED_NODE_AFTER
+    ]:
+        if helper_config.get(field):
+            try:
+                helper_config[field] = int(helper_config[field])
+            except ValueError:
+                raise WazuhError(3004, extra_message=f"HAProxy Helper {field} must be an integer.")
+
```

Comment:
We could also validate that the `HAPROXY_PROTOCOL` is either `http` or `https` (the C code does this, I don't know if it's necessary). And verifying that the `HAPROXY_ADDRESS` is a valid IPv4 or IPv6 address.

Reply:
I decided to keep the validation, for `HAPROXY_PROTOCOL`, in the C code to not duplicate it.
Regarding `HAPROXY_ADDRESS` in the previous JSON schema it was just a string, so I decided to keep the same approach because it can be a hostname or an IP address. In any case, if the provided address is wrong an error 3043 will be raised.
```python
2024/04/10 18:20:08 CRITICAL: [HAPHelper] [Main] Cannot initialize the proxy
2024/04/10 18:20:08 CRITICAL: [HAPHelper] [Main] Error 3043 - Cannot initialize Proxy API: Check connectivity and the configuration file
2024/04/10 18:20:08 CRITICAL: [HAPHelper] [Main] Unexpected exception: Error 3043 - Cannot initialize Proxy API: Check connectivity and the configuration file
    raise WazuhHAPHelperError(3043, extra_message='Check connectivity and the configuration file')
wazuh.core.exception.WazuhHAPHelperError: Error 3043 - Cannot initialize Proxy API: Check connectivity and the configuration in the `ossec.conf`
2024/04/10 18:20:08 INFO: [HAPHelper] [Main] Process ended
```

Reply:
@Selutario do you see any problem keeping the validations in `validate_haproxy_helper_config`? Obviously adding more.

Before passing the config to `check_cluster_config` the config was loaded with `read_config` so already was validated. https://github.com/wazuh/wazuh/blob/04781054cff4e8758a283c684372c5b431e9b14b/framework/scripts/wazuh_clusterd.py#L204

Reply:
I decided to split the process into two steps. The parse of the configuration and the validation.

For example, with the given configuration:

```xml
<cluster>
    <name>wazuh</name>
    <node_name>master-node</node_name>
    <node_type>master</node_type>
    <key>9d273b53510fef702b54a92e9cffc82e</key>
    <port>1516</port>
    <bind_addr>0.0.0.0</bind_addr>
    <nodes>
        <node>wazuh-master</node>
    </nodes>
    <hidden>no</hidden>
    <disabled>no</disabled>
    <haproxy_helper>
        <haproxy_disabled>no</haproxy_disabled>
        <haproxy_address>wazuh-proxy</haproxy_address>
        <haproxy_user>haproxy</haproxy_user>
        <haproxy_password>haproxy</haproxy_password>
        <excluded_nodes>master-node,worker1,worker2</excluded_nodes>
        <haproxy_protocol>http</haproxy_protocol>
        <frequency>1</frequency>
    </haproxy_helper>
  </cluster>
```

The check cluster config will raise the exception for minimum value:

```python
2024/04/22 14:53:29 ERROR: [Cluster] [Main] Error 3004 - Error in cluster configuration: Invalid value for frequency. 1 is less than the minimum of 10
Traceback (most recent call last):
  File "/var/ossec/framework/python/lib/python3.10/site-packages/wazuh/core/cluster/cluster.py", line 79, in validate_haproxy_helper_config
    validate(config, SCHEMA)
  File "/var/ossec/framework/python/lib/python3.10/site-packages/jsonschema/validators.py", line 541, in validate
    cls(schema, *args, **kwargs).validate(instance)
  File "/var/ossec/framework/python/lib/python3.10/site-packages/jsonschema/validators.py", line 130, in validate
    raise error
jsonschema.exceptions.ValidationError: 1 is less than the minimum of 10

Failed validating 'minimum' in schema['properties']['frequency']:
    {'minimum': 10, 'type': 'integer'}

On instance:
    1

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/var/ossec/framework/scripts/wazuh_clusterd.py", line 216, in main
    wazuh.core.cluster.cluster.check_cluster_config(cluster_configuration)
  File "/var/ossec/framework/python/lib/python3.10/site-packages/wazuh/core/cluster/cluster.py", line 137, in check_cluster_config
    validate_haproxy_helper_config(config.get(HAPROXY_HELPER, {}))
  File "/var/ossec/framework/python/lib/python3.10/site-packages/wazuh/core/cluster/cluster.py", line 81, in validate_haproxy_helper_config
    raise WazuhError(
wazuh.core.exception.WazuhError: Error 3004 - Error in cluster configuration: Invalid value for frequency. 1 is less than the minimum of 1
```